lib/game.js

```diff
@@ -0,0 +1,52 @@
+function Game() {
+  this.frames = [];
+  MAX_FRAMES = 10;
+};
+
+Game.prototype = {
+  getFrames: function () {
+    return this.frames
+  },
+
+  currentFrame: function() {
+    return this.frames[this.frames.length - 1];
+  },
+
+  currentFrameNo: function () {
+    return this.frames.length
+  },
+
+  addFrame: function (frame) {
+    this.frames.push(frame)
+  },
+
+  addRoll: function (pins) {
+    var frameNumber = this.currentFrameNo()
+    if (this.currentFrame().isFull(frameNumber)) {
+      if (frameNumber === MAX_FRAMES) { return } 
+      this.addFrame(new Frame())
+    }
+    this.currentFrame().addRoll(pins)
+  },
+
+  score: function () {
+    return this.getFrames().reduce(
+      (total, frame, index) => total + frame.score() + this.bonusScore(frame, index),
+      0
+    );
+  },
+
+  // works but needs some refactoring
+  bonusScore: function (frame, index) {
```

Comment:
It feels to me like this should be a factor of the frame. Rather than calculating the bonus score here and holding that state outside the frame, why not tell the preceding frames 'here's a new roll' and let the frame decide whether it is waiting for a bonus? That way you can keep all of the frame score logic inside the frame. 

The fact that you struggled to a isolate your Game class (as you mention in your self review) is a side effect of your approach here - your Game is too tightly coupled with your Frame. Whenever you find yourself struggling to isolate logic for testing, ask yourself if you've really got the right design and consider entering a refactor step.